File: app/src/main/java/com/nobrain/samples/daumeimagesearch/home/presenter/HomePresenter.java
package com.nobrain.samples.daumeimagesearch.home.presenter;

public interface HomePresenter {

    void inputSearchText(String searchText);

    void unSubscribeSearch();

    void onItemClick(int position);

    interface View {

        void refresh();

        void onMoveLink(String link);
    }
}

File: app/src/main/java/com/nobrain/samples/daumeimagesearch/home/presenter/HomePresenter.kt
package com.nobrain.samples.daumeimagesearch.home.presenter

interface HomePresenter {

    fun inputSearchText(searchText: String)

    fun unSubscribeSearch()

    fun onItemClick(position: Int)

    interface View {

        fun refresh()

        fun onMoveLink(link: String)
    }
}
